Implementing bitcode.

35C4n0r opened the pull request:
This PR Implements the pure python variant of intbitset.

Class:
- [x] intbitset

Methods:
- [x] add
- [x] clear
- [x] copy
- [x] difference
- [x] difference_update
- [x] discard
- [ ] extract_finite_list
- [ ] fastdump
- [ ] fastload
- [ ] get_allocated
- [ ] get_size
- [ ] get_wordbitsize
- [ ] get_wordbytesize
- [x] intersection
- [x] intersection_update
- [x] isdisjoint
- [x] issubset
- [x] issuprset
- [ ] is_infinite
- [ ] pop
- [x] remove
- [x] strbits
- [x] symetric_difference
- [x] symetric_difference_update
- [x] tolist
- [x] union
- [x] union_update
- [ ] update
- [ ] update_with_signs

35C4n0r committed:
add: added intbitset.py
- added intbitset class and it's methods.

Signed-off-by: 35C4n0r <jaykumar20march@gmail.com>

35C4n0r committed:
add: added remaining methods
- added the remaining methods for intbitset class.

Signed-off-by: 35C4n0r <jaykumar20march@gmail.com>

JonoYang commented on src/bitcode/intbitset.py:
```diff
@@ -0,0 +1,194 @@
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# ahocode is a trademark of nexB Inc.
```
```suggestion
# bitcode is a trademark of nexB Inc.

```

JonoYang commented on src/bitcode/intbitset.py:
```diff
@@ -0,0 +1,194 @@
+# ahocode is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/nexB/ahocode for support or download.
```
```suggestion
# See https://github.com/nexB/bitcode for support or download.

```

JonoYang commented on src/bitcode/intbitset.py:
```diff
@@ -0,0 +1,194 @@
+    def copy(self):
+        """ Return a shallow copy of a set. """
+        new = intbitset()
+        intbitset.bitset = self.bitset
```
Should this be `new.bitset = self.bitset`?

JonoYang reviewed (changes requested):
@35C4n0r Just pointing out a few small things

35C4n0r committed:
Update src/bitcode/intbitset.py

Refactoring Typos.

Co-authored-by: Jono Yang <JonoYang@users.noreply.github.com>

35C4n0r committed:
Update src/bitcode/intbitset.py

Refactoring Typos.

Co-authored-by: Jono Yang <JonoYang@users.noreply.github.com>

35C4n0r committed:
add: added tests and GitHub workflow
- added new tests for the intbitset class.
- updated README.rst.
- Tried adding workflows.

Signed-off-by: 35C4n0r <jaykumar20march@gmail.com>

35C4n0r committed:
ref: minor refactoring in comments.

Signed-off-by: 35C4n0r <jaykumar20march@gmail.com>

35C4n0r commented:
@JonoYang, thanks for the review. I have added the changes, a few tests, and I also tried adding the github workflow to run the test.

JonoYang commented:
@35C4n0r I've enabled Azure Pipelines for CI testing on the ahocode and bitcode repos

35C4n0r committed:
ref: removed GitHub workflows.

Signed-off-by: 35C4n0r <jaykumar20march@gmail.com>

35C4n0r commented:
@JonoYang I have removed the github workflows, and i don't thin i need to make any changes in the azure-pipelines.yml
The PR is ready for review.

35C4n0r commented:
@JonoYang also please make a repo for lxml so that I can start working on it.

JonoYang commented:
@35C4n0r I've created the xml fallback library repo at https://github.com/nexb/sanexml

JonoYang commented:
@35C4n0r Is there anything else you need to add to this PR, otherwise I will merge it.

35C4n0r commented:
@JonoYang, I don't have anything more to add to PR.

JonoYang commented:
@35C4n0r I will merge the PR. Thanks for the contributions!
